### PR TITLE
boards/sodaq: factorize common code

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR is factorizing the common code of SODAQ boards.

Peripherals added to common configuration code: clock, rtc, rtt, spi, timer, usbdev.
Others are different (uart, i2c, adc) or not implemented for all boards (pwm).

### Testing procedure

Verify features still works on SODAQ boards.

### Issues/PRs references

None

### CODE REVIEW

**Comment by keestux** [comment]
First thing to notice is that not all boards have the same SAMD21 variant. Two have SAMD21G18A and two have SAMD21J18A.

**Comment by aabadie** [comment]
> First thing to notice is that not all boards have the same SAMD21 variant. Two have SAMD21G18A and two have SAMD21J18A.

Indeed, but only sodaq-one has a samd21g18a. Thanks for pointing that out, I hope there are no other _little_ mistakes like this.

File: boards/sodaq-explorer/include/periph_conf.h [hunk 1]
```diff
@@ -190,27 +105,6 @@ static const adc_conf_chan_t adc_channels[] = {
...
- * @{
- */
-static const spi_conf_t spi_config[] = {
-    {
-        .dev      = &SERCOM3->SPI,
-        .miso_pin = GPIO_PIN(PA, 22),
-        .mosi_pin = GPIO_PIN(PA, 20),
-        .clk_pin  = GPIO_PIN(PA, 21),
-        .miso_mux = GPIO_MUX_C,
-        .mosi_mux = GPIO_MUX_C,
```
Comment (aabadie) [inline boards/sodaq-explorer/include/periph_conf.h]: I think there was an existing mistake on pin MUX for MOSI and CLK.

**Comment by keestux** [comment]
@aabadie first of all, thanks for doing this. It is really a good move to make things common.

I want to spend a bit of time to verify the changes. But since I'm doing all this in my spare time, it will take a few days.

**Comment by aabadie** [comment]
I've just double checked the other peripheral configuration and couldn't find any other mistakes there.

**Comment by aabadie** [comment]
> But since I'm doing all this in my spare time, it will take a few days.

No problem, take your time. We are in feature freeze so there's no emergency for merging this.

**Comment by keestux** [comment]
>     First thing to notice is that not all boards have the same SAMD21
>     variant. Two have SAMD21G18A and two have SAMD21J18A.
>
> Indeed, but only sodaq-one has a samd21g18a. Thanks for pointing that 
> out, I hope there are no other /little/ mistakes like this.
>
Correct, one SAMD21G18A and three SAMD21J18A. But there is another board 
waiting to be added with a SAMD21G18A. Adding that board will be easier 
with this PR.

File: boards/sodaq-explorer/include/periph_conf.h (hunk 1 above)
Comment (keestux) [reply boards/sodaq-explorer/include/periph_conf.h]: Indeed

**Comment by keestux** [comment]
@aabadie Besides the `CPU_MODEL` I could not find anything else.
If you make the CPU_MODEL change I can finish the review

**Comment by aabadie** [comment]
> If you make the CPU_MODEL change I can finish the review

if it's the one about samd21g18a, then it's already done in the last 2 fixup commits.

**Pull request merged**